lnwallet/commitment.go

```diff
@@ -514,37 +514,87 @@ func HtlcSuccessFee(chanType channeldb.ChannelType,
 
 // CommitScriptAnchors return the scripts to use for the local and remote
 // anchor.
-func CommitScriptAnchors(localChanCfg,
-	remoteChanCfg *channeldb.ChannelConfig) (*ScriptInfo,
+func CommitScriptAnchors(chanType channeldb.ChannelType,
+	localChanCfg, remoteChanCfg *channeldb.ChannelConfig) (*ScriptInfo,
 	*ScriptInfo, error) {
 
-	// Helper to create anchor ScriptInfo from key.
-	anchorScript := func(key *btcec.PublicKey) (*ScriptInfo, error) {
-		script, err := input.CommitScriptAnchor(key)
-		if err != nil {
-			return nil, err
+	var (
+		anchorScript func(key *btcec.PublicKey) (*ScriptInfo, error)
+		keySelector  func(*channeldb.ChannelConfig, bool) *btcec.PublicKey
+	)
+
+	switch {
+	// For taproot channels, the anchor is slightly different: the top
+	// level key is now the (relative) local delay and remote public key,
+	// since these are fully revealed once the commitment hits the chain.
+	//
+	// TODO(roasbeef): need to re-examine the assumption of what's
+	// revealed? otherwise then have two levels of tweaks...
+	case chanType.IsTaproot():
+		anchorScript = func(key *btcec.PublicKey) (*ScriptInfo, error) {
+			anchorKey, err := input.TaprootOutputKeyAnchor(key)
+			if err != nil {
+				return nil, err
+			}
+
+			anchorPkScript, err := input.PayToTaprootScript(anchorKey)
+			if err != nil {
+				return nil, err
+			}
+
+			return &ScriptInfo{
+				PkScript: anchorPkScript,
+			}, nil
 		}
 
-		scriptHash, err := input.WitnessScriptHash(script)
-		if err != nil {
-			return nil, err
+		keySelector = func(cfg *channeldb.ChannelConfig,
+			local bool) *btcec.PublicKey {
+
+			if local {
+				return cfg.DelayBasePoint.PubKey
```

Comment:
I think this needs to be tweaked by the commitpoint. Otherwise, it's not claimable by a 3rd party after 16 blocks since the to_local output uses the tweaked `DelayBasePoint`.

Reply:
Good point, thinking about it a bit more, this actually needs to be the keys that's _revealed_ on chain. So then this would be the top level tweaked keys for each output. 

If we do this though, then we end up w/ that "double" tweak situation I was trying to avoid in the past (private key tweaked twice for the keyspend path). 

Updated for now to use the tweaked key.